### ISSUE TITLE
[question] Run command after checkout and before docker build

### ISSUE THREAD

**Issue by gmmcal** [GoogleContainerTools/kaniko#2986]
Hello,

tl;dr: is there a way for me to run a shell command after `git checkout` and before `docker build` in kaniko?

I have kind of an odd problem and I am trying to work around a solution. 

I work for a company and my team provides paved road for application (aka templates). Due to a historical decision, all apps relies on a specific file to be created right before `docker build` step on our pipelines (currently jenkins). While migrating such pipelines to run on GH Actions and using kaniko, that step to create file should die. That is great, but it is where problem starts. My team supports a considerable amount of teams and each team supports 1:n applications, most of them in the need of this file to be present and this will cause migration to take much longer than it could.

The only way I could work around that limitation is create that file manually and put it on `/kaniko/buildcontext/` folder, which seems to be the folder code is being checked out by kaniko. But, if I create that file before I run `/kaniko/executor` and then run `/kaniko/executor --context="${{ github.repositoryUrl }}#${{ github.sha }}" <more-params>`, my original file is gone and my build fails because file is not there.

Is there a way for me to run a shell command after `git checkout` and before `docker build` in kaniko? something like a hook? or maybe a way that kaniko would not checkout code and then I would do it myself somehow? if yes, how? looks like kaniko image `gcr.io/kaniko-project/executor:debug` does not include `git` client and I have no idea how to work around this at this moment.

Thanks

**Comment by gmmcal** [comment]
Update: I managed to work around this topic by evaluating all parameters ahead of time and sending to build step as build arguments.

Personally, I still think that having a way to attach "hooks" to before and post build could be an interesting feature.

For now, I will close this issue.
